### PR TITLE
MrackConfig: improve the approach to be object based and easy to extend 

### PR DESCRIPTION
**Pull request by pvoborni**
base on PR: https://github.com/neoave/mrack/pull/36

This commit implements a MrackConfig which allows to define options
and arguments otherwise needed in a configuration file and thus
save people's time with writing them in CLI.

Config file looks like:
```
[mrack]
mrackdb = config/mrackdb.yaml
provisioning-config = config/provisioning-config.yaml
metadata = config/metadata.yaml
```

Where each value is optional and existance of the config is also
optional.

Mrack looks for the config on these locations:
- user provided path via -c or --mrack-config global option
- ./mrack.conf
- ~/.mrack/mrack.conf
- /etc/mrack/mrack.conf

It also changes CLI of commands:
- metadata is no longer an argument but an option: -m or --metadata
- provisioning config shortcut changes to global -p

With these changes it is possible to define a config file to allow
just typing:
```
$ mrack up
$ mrack list
$ mrack ssh
$ mrack output
$ mrack destroy
```

The MrackConfig class is also extensible for new use cases.

Signed-off-by: Petr Vobornik <pvoborni@redhat.com>

### CODE REVIEW

File: src/mrack/data/mrack.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[mrack]
+mrackdb = .mrackdb.json
+provisioning-config = provisioning-config.yaml
+
```
Comment (Tiboris) [inline src/mrack/data/mrack.conf:4]: Could you please add here `metadata` key value example?

Comment (Tiboris) [reply src/mrack/data/mrack.conf:4]: > Mrack looks for the config on these locations:
>
>    user provided path via -c or --mrack-config global option
>    ~./mrack.conf

I believe `~` should not be there

Comment (pvoborni) [reply src/mrack/data/mrack.conf]: Commit message ammended.

Comment (pvoborni) [reply src/mrack/data/mrack.conf]: And metadata added to data/mrack.conf

**Comment by Tiboris** [comment]
Works for me @pvoborni thanks for the patch and improvement :)

**Review by Tiboris — APPROVED** [review]
Works for me @pvoborni thanks for the patch and improvement :)

**Pull request merged**